cli/tests/integration/vendor_tests.rs

```diff
@@ -537,6 +538,51 @@ fn update_existing_config_test() {
   assert!(output.status.success());
 }
 
+#[test]
+fn vendor_npm_node_specifiers() {
+  let context = TestContextBuilder::for_npm().use_temp_cwd().build();
+  let temp_dir = context.temp_dir();
+  temp_dir.write(
+    "my_app.ts",
+    concat!(
+      "import { path, getValue, setValue } from 'http://localhost:4545/vendor/npm_and_node_specifier.ts';\n",
+      "setValue(5);\n",
+      "console.log(path.isAbsolute(Deno.cwd()), getValue());",
+    ),
+  );
+  temp_dir.write("deno.json", "{}");
+
+  let output = context.new_command().args("vendor my_app.ts").run();
+  output.assert_matches_text(
+    format!(
+      concat!(
+        "Download http://localhost:4545/vendor/npm_and_node_specifier.ts\n",
+        "Download http://localhost:4545/npm/registry/@denotest/esm-basic\n",
+        "Download http://localhost:4545/npm/registry/@denotest/esm-basic/1.0.0.tgz\n",
+        "{}\n",
+      ),
+      success_text_updated_deno_json("1 module", "vendor/")
+    )
+  );
+  let output = context.new_command().args("run -A my_app.ts").run();
+  output.assert_matches_text("Initialize @denotest/esm-basic@1.0.0\ntrue 5\n");
```

Comment:
Hmmm... actually, this shouldn't require a run or cache to then vendor. Might be complicated though.